### PR TITLE
Fix code coverage

### LINKED ISSUE #96: Fix code coverage
As @bobheadxi pointed out, our code coverage appears to only include the `tests/` directory (hence why we have perfect 100% code coverage!). It should include everything.

### PR DESCRIPTION
**Pull request by rwblickhan**
# Pull Request

## Description

This updates the `build_check.sh` script to always produce code coverage for the _whole_ repo, not just the `tests/` directory, and updates the Travis CI config to use the script instead of its own version.

It also enables branch coverage, though I don't believe Codecov actually makes use of this.

## Testing

See [Codecov](https://codecov.io/gh/ubclaunchpad/rocket2.0/branch/rwblickhan%2F%2396-fix-codecov) for the coverage produced by this branch and [Travis CI](https://travis-ci.org/ubclaunchpad/rocket2.0/builds/454267754) to see the corrected build.

## Ticket(s)

Closes #96 

(Create a copy of that line for each Github Issue affected,
and replace "Affects" with "Closes" if merging this will close the relevant ticket.)

### CODE REVIEW

**Comment by cheukyin699** [comment]
Disable codecov for a moment? We're really close (but not hitting) the 3% mark.

**Comment by rwblickhan** [comment]
I'm going to add a unit test or two for the Bot class - if you look at it [on Codecov](https://codecov.io/gh/ubclaunchpad/rocket2.0/src/1a2c3038b2ba0564ed2230fc6936229be55573e0/bot/bot.py) we don't have tests for when the API call fails, which should be easy to add and put us back over.

File: tests/bot/bot_test.py [hunk 1]
```diff
@@ -18,10 +18,26 @@ def test_send_dm(self):
...
+    def test_send_dm_failure(self):
+        """Test send_dm() when the Slack API call fails."""
+        bot = Bot(self.mock_sc)
+        self.mock_sc.api_call = mock.MagicMock(return_value={"error": "Error"})
+
+        try:
+            bot.send_dm("Hahahaha", "UD8UCTN05")
+        except SlackAPIError as e:
+            assert e.error == "Error"
+        finally:
```
Comment (cheukyin699) [inline tests/bot/bot_test.py]: First time I've ever seen anyone use `finally`!

Comment (rwblickhan) [reply tests/bot/bot_test.py]: Yeah, not sure it’s actually useful here... but it does signal intent, so 🤷‍♀️

**Pull request merged**